Project.java

```diff
@@ -18,60 +40,38 @@ class Food implements Serializable
     {
         this.itemno=itemno;
         this.quantity=quantity;
-        switch(itemno)
-        {
-            case 1:price=quantity*50;
-                break;
-            case 2:price=quantity*60;
-                break;
-            case 3:price=quantity*70;
-                break;
-            case 4:price=quantity*30;
-                break;
-        }
+        this.price=FoodList.foodList.get(itemno-1).unitPrice*quantity;
     }
 }
-class Singleroom implements Serializable
-{
-    String name;
-    String contact;
-    String gender;   
-    ArrayList<Food> food =new ArrayList<>();
+abstract class Room implements Serializable{
+	ArrayList<Guest> guests=new ArrayList<>();
+	int maxCapacity;
+	ArrayList<Food> food =new ArrayList<>();
+	
+	void addGuest(Guest g) {
+		if(g!=null && guests.size()<maxCapacity)
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/shouryaj98/Hotel-Management-Project-Java/pull/4#issuecomment-2437657272>#2aa38a</a><span> - 10/25/2024, 12:36 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> Potential performance issue in the addGuest method due to unnecessary null check and redundant condition
</div>
<div id="fix">
<b>Fix:</b> Optimize the addGuest method by removing the redundant null check and simplifying the condition
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -51,7 +51,7 @@ abstract class Room implements Serializable{
  	void addGuest(Guest g) {
 -		if(g!=null && guests.size()<maxCapacity)
 +		if(guests.size() < maxCapacity)
  			guests.add(g);
 -		else if(g!=null)
 +		else
  			throw new NotAvailable();
 		
  	}
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged